src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1610,10 +1634,10 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
                 }
 
                 // May have media/type; charset=utf-8, allow partial match.
-                if (!Request.ContentType.StartsWith("application/x-www-form-urlencoded", StringComparison.OrdinalIgnoreCase)) {
+                if (Request.ContentType.IndexOf("application/x-www-form-urlencoded", StringComparison.OrdinalIgnoreCase) != -1) {
```

Comment:
The media type is `application/x-www-form-urlencoded`. Any (optional) subtype would have to be inserted after the main type: https://www.ietf.org/rfc/rfc2045.txt

FYI: this check is the same as the one used by the OIDC client middleware: https://github.com/aspnet/Security/blob/dev/src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs#L318
If you think it is wrong, I suggest opening a new ticket on the aspnet/Security repository. In the meantime, consider reverting this change.